### DISCUSSION TITLE
Syntax highlighting for wbt and proto files in external editors

### DISCUSSION THREAD

**Discussion by PeterJCLaw** [cyberbotics/webots#6408] [Ideas]
Are there any good solutions for syntax highlighting (and other editor support) of `.wbt` and `.proto` files in external editors? (Currently I end up picking another language which has vaguely similar syntax and pretending it's that).

I've created VSCode extensions before so would be up for trying to create one, probably just starting with basic syntax highlighting, though if there's an existing spec of the language somewhere (ideally in `.tmLanguage` format or similar) that would be a great jumping off point.

**Comment by omichel** [comment]
Check [this](https://github.com/tn12787/PROTO-Webots) out. I am not sure however, if it is still up-to-date and functional.

**Comment by PeterJCLaw** [comment]
Thanks, that's a handy reference. I've ended up not actually using much of what that extension contains, but it was useful to have examples of some things.

I've got a basic extension at https://github.com/PeterJCLaw/vscode-webots. Not published anywhere yet, but seems to mostly work:

![image](https://github.com/cyberbotics/webots/assets/336212/a583c65c-946c-49dd-b72b-e14ba49ae4ee)

**Comment by PeterJCLaw** [comment]
Turns out that https://marketplace.visualstudio.com/items?itemName=kagamma.cge-x3d-vscode also exists, which seems more complete than mine. Will have a play with that.
